mlflow/tracing/trace_manager.py

```diff
@@ -198,11 +208,22 @@ def get_root_span_id(self, request_id) -> Optional[str]:
 
         if trace:
             for span in trace.span_dict.values():
-                if span.parent_span_id is None:
+                if span.parent_id is None:
                     return span.span_id
 
         return None
 
+    def get_or_create_request_id(self, trace_id: str) -> Optional[str]:
+        """
+        Get the request ID for the given trace ID. If the request ID does not exist, create a new
+        request ID and return it.
+        """
+        if request_id := self._trace_id_to_request_id.get(trace_id):
+            return request_id
+
+        # TODO: Request ID should be generated by the backend and fetched via StartTrace API.
+        return f"tr-{trace_id}"
```

Comment:
We might want to discuss the meaning of request_id 🤔 Should it still start with `tr-` prefix and why do we need it, seems trace-id is also a unique identifier for a trace. (not for this PR though)